src/CTA.Rules.Analysis/RulesAnalysis.cs

```diff
@@ -234,6 +234,16 @@ private bool AnalyzeChildren(FileActions fileAction, UstList<UstNode> children,
                                             overrideKey = compareToken.Key;
                                         }
                                     }
+                                    //If the semanticClassType is too specific to apply to all TData types
+                                    if(token == null)
+                                    {
+                                        if(invocationExpression.SemanticClassType.Contains('<'))
+                                        {
+                                            string semanticClassType = invocationExpression.SemanticClassType.Substring(0,invocationExpression.SemanticClassType.IndexOf('<'));
+                                            compareToken = new InvocationExpressionToken() { Key = invocationExpression.SemanticOriginalDefinition, Namespace = invocationExpression.Reference.Namespace, Type = semanticClassType };
```

Comment:
Not a necessary change, but two suggestions to make this more readable/give additional context to readers (assuming my understanding is correct):
1. Replace `invocationExpression.SemanticClassType.Contains('<')` with a function called `IsGenericType(string semanaticClassType)` or something similar
2. Replace `invocationExpression.SemanticClassType.Substring(0,invocationExpression.SemanticClassType.IndexOf('<'))` with a function called `RemoveTypeParametersFromGenericType(string semanticClassType)` or something similar